containers/orchestration/app/services.py

```
@@ -81,6 +90,31 @@ def message_parser_payload(**kwargs) -> dict:
     return data
 
 
+def save_to_db(**kwargs) -> dict:
+    ecr_id = kwargs["payload"]["ecr_id"]
+    payload_data = kwargs["payload"]["data"]
+    url = kwargs["url"]
+    engine = create_engine(url)
+    pg_data = PostgresFhirDataModel(ecr_id=str(ecr_id), data=payload_data)
+    try:
+        with Session(engine, expire_on_commit=False) as session:
+            repo = SqlAlchemyFhirRepository(session)
+            repo.persist(pg_data)
+        return CustomJSONResponse(content=jsonable_encoder(payload_data), url=url)
+    except SQLAlchemyError as e:
+        return Response(content=e, status_code=500)
+
+
+def save_to_db_payload(**kwargs) -> dict:
+    response = kwargs["response"]
+    r = response.json()
+    if r.get("parsed_values", {}).get("eicr_id"):
+        ecr_id = r["parsed_values"]["eicr_id"]
+    else:
+        ecr_id = uuid.uuid4()
```

I'd actually argue there should be different behavior here - if the eicr ID can't be found, I think the service should throw an exception rather than generate a random UUID.

The product thinking here is that we'll only be able to find the eCRs users are looking at in NBS through the original eICR ID - they won't know what the generated ID is.

---

Good idea, added an exception here and a test for it